build_projects/update-dependencies/Program.cs

```diff
@@ -1,25 +1,93 @@
+            buildInfos.Add(BuildInfo.Get("CoreSetup", s_config.CoreSetupVersionUrl, fetchLatestReleaseFile: false));
+
+            IEnumerable<IDependencyUpdater> updaters = GetUpdaters();
+
+            GitHubAuth gitHubAuth = new GitHubAuth(s_config.Password, s_config.UserName, s_config.Email);
+
+            DependencyUpdater dependencyUpdater = new DependencyUpdater(
+                gitHubAuth,
+                s_config.GitHubProject,
+                s_config.GitHubOriginOwner,
+                s_config.GitHubUpstreamBranch,
+                s_config.UserName,
+                s_config.GitHubPullRequestNotifications);
+
+            if (args.Length > 0 && string.Equals("--Update", args[0], StringComparison.OrdinalIgnoreCase))
+            {
+                dependencyUpdater.Update(updaters, buildInfos);
+            }
+            else
+            {
+                dependencyUpdater.UpdateAndSubmitPullRequestAsync(updaters, buildInfos);
+            }
+        }
+
+        private static IEnumerable<IDependencyUpdater> GetUpdaters()
+        {
+            yield return CreateProjectJsonUpdater();
+
+            yield return CreateRegexUpdater(@"build_projects\shared-build-targets-utils\DependencyVersions.cs", "CoreCLRVersion", "Microsoft.NETCore.Runtime.CoreCLR");
+            yield return CreateRegexUpdater(@"build_projects\shared-build-targets-utils\DependencyVersions.cs", "JitVersion", "Microsoft.NETCore.Jit");
+
+            yield return CreateRegexUpdater(@"build_projects\dotnet-cli-build\CliDependencyVersions.cs", "SharedFrameworkVersion", "Microsoft.NETCore.App");
+            yield return CreateRegexUpdater(@"build_projects\dotnet-cli-build\CliDependencyVersions.cs", "HostFxrVersion", "Microsoft.NETCore.DotNetHostResolver");
+            yield return CreateRegexUpdater(@"build_projects\dotnet-cli-build\CliDependencyVersions.cs", "SharedHostVersion", "Microsoft.NETCore.DotNetHost");
+        }
+
+        private static IDependencyUpdater CreateProjectJsonUpdater()
+        {
+            const string noUpdateFileName = ".noautoupdate";
+
+            IEnumerable<string> projectJsonFiles = Enumerable.Union(
+                Directory.GetFiles(Dirs.RepoRoot, "project.json", SearchOption.AllDirectories),
+                Directory.GetFiles(Path.Combine(Dirs.RepoRoot, @"src\dotnet\commands\dotnet-new"), "project.json.template", SearchOption.AllDirectories))
+                .Where(p => !File.Exists(Path.Combine(Path.GetDirectoryName(p), noUpdateFileName)) &&
+                    !Path.GetDirectoryName(p).EndsWith("CSharp_Web", StringComparison.Ordinal));
+
+
+            return new ProjectJsonUpdater(projectJsonFiles)
+            {
+                SkipStableVersions = false
+            };
+        }
+
+        private static IDependencyUpdater CreateRegexUpdater(string repoRelativePath, string dependencyPropertyName, string packageId)
+        {
+            return new FileRegexPackageUpdater()
```

Comment:
More succinct, less things for the eye to scan through. Cleaner. There. A few reason.